lib/sources/service.rb

```diff
@@ -0,0 +1,27 @@
+module Sources
+  class Service
+    def self.call(klass)
+      setup
+      yield init(klass)
+    rescue SourcesApiClient::ApiError => err
+      Rails.logger.error("SourcesApiClient::ApiError #{err.message} ")
+      raise Catalog::SourcesError, err.message
+    end
+
+    private_class_method def self.setup
+      SourcesApiClient.configure do |config|
+        config.host   = ENV['SOURCES_URL'] || 'localhost'
+        config.scheme = URI.parse(ENV['SOURCES_URL']).try(:scheme) || 'http'
+        dev_credentials(config)
+      end
+    end
+
+    private_class_method def self.init(klass)
+      headers = ManageIQ::API::Common::Request.current_forwardable
+      Rails.logger.info("Sending Headers to Sources #{headers}")
```

Comment:
Yes it's what we're using for the RBAC API and approval API. I'm debating changing it to the way the other ones are though only because of the fact that we only have one API class for Sources (`SourcesApiClient::DefaultApi`).